addons/mail/static/src/messaging/entity/activity/activity.js

```diff
@@ -105,6 +115,9 @@ function ActivityFactory({ Entity }) {
                     activityTypeId,
                     activityTypeDisplayName,
                 ] = [],
+                // FIXME both following fields should be handled in Approvals
+                approver_id: approverId,
+                approver_status: approverStatus,
```

Comment:
I don't know how you could factorize these two calls in one as they are not totally identical. One of them (in activity) make one RPC call to update the current activity, the second one make one RPC call to update all activities linked to the chatter. The only thing in common here is insert/update method which is difficult to override for the moment ... So am I right to think that we should make an override of `chatter.refreshActivities` and `activity.fetchAndUpdate` (and these should become class methods because of that ... ) ?

Reply:
Trying things and thinking about that and I think we need an override of `_update` method because we need to handle the data from server, or you put a static method in activity which call  _update ? But then you'll need to override insert too...

So I'll try to override static `create` method